04/app.js

```diff
@@ -0,0 +1,37 @@
+const divs = document.querySelectorAll("div");
+const body = document.querySelector("body");
+
+
+const myFunction = function (event, element) {
+    
+    element = this;
+    //event.stopPropagation();
+    setTimeout(function() {element.classList.add("clicked")}, element.dataset.time)
+    if (divs[2].className.includes('clicked') === true) {
+        console.log("dziala")
+        event.stopPropagation();
+        /*divs.forEach(function(element) {
+            element = this;
+            setTimeout(function() {
+                element.classList.add("clicked")}, element.dataset.time)
+        }); */
+    }
+}
+divs.forEach(function(element) {
+    element.addEventListener("click", myFunction)
+});
+
+
+
+const secondFunction = function () {
+
```

Comment:
Tutaj można by było sprawdzić czy `e.target === e.currentTarget` to pozwala określić czy event jest uruchomiony przez propagację czy przez bezpośrednie kliknięcie w element ;)